packages/shylu/shylu_dd/bddc/src/ProblemMakerBDDC.h

```diff
@@ -78,39 +92,75 @@ template <class LO, class GO, class SX, class SM> class ProblemMaker
     m_Comm(Comm),
     m_problemType(Parameters->get("Problem Type", SCALARPDE)),
     m_analysisType(Parameters->get("Analysis Type", STANDARD)),
-    m_spatialDim(Parameters->get("Spatial Dimension", 1))
+    m_spatialDim(Parameters->get("Spatial Dimension", 1)),
+    m_readModel(false)
   {
     MPI_Comm_size(Comm, &m_numProc);
     MPI_Comm_rank(Comm, &m_myPID);
     GenerateModel();
     ApplyEssentialBCs();
-    GenerateConstraintEquations();
+    setQuadraturePoints();
+  }
+
+  ProblemMaker(RCP<Teuchos::ParameterList> Parameters,
+	       MPI_Comm Comm,
+	       const std::vector<SM> & xFile,
+	       const std::vector<SM> & yFile,
+	       const std::vector<SM> & zFile,
+	       const std::vector< std::vector<LO> > & elemConn,
+	       const std::vector<GO> & nodeGIDs,
+	       const std::vector< std::vector<LO> > & subElems,
+	       const std::vector<LO> & constrainedNodes) :
+  m_Parameters(Parameters),
+    m_Comm(Comm),
+    m_problemType(Parameters->get("Problem Type", SCALARPDE)),
+    m_analysisType(Parameters->get("Analysis Type", STANDARD)),
+    m_spatialDim(Parameters->get("Spatial Dimension", 1)),
+    m_numNode(nodeGIDs.size()),
+    m_numElem(elemConn.size()),
+    m_x(xFile),
+    m_y(yFile),
+    m_z(zFile),
+    m_nodeGIDs(nodeGIDs),
+    m_elemConn(elemConn),
+    m_readModel(true)
+  {
+    MPI_Comm_size(Comm, &m_numProc);
+    MPI_Comm_rank(Comm, &m_myPID);
+    determineNodeBegLocalDof();
+    m_subElems = subElems;
+    ApplyEssentialBCs(constrainedNodes);
+    setQuadraturePoints();
   }
 
   ~ProblemMaker()
   {
+#ifdef TENSORPRODUCTBDDC
+    delete m_TP;
```

Comment:
Consider using `std::unique_ptr` / `std::shared_ptr` (whichever is appropriate), instead of a raw pointer.

Reply:
I'll plan to do this when I submit another pull request that includes new content for higher-order elements.

Reply:
We discussed using `std::unique_ptr`, and I showed a code example with `std::unique_ptr<int[]>`.